pkg/operation/botanist/health_check.go

```diff
@@ -383,7 +383,9 @@ func (b *HealthChecker) FailedCondition(condition gardencorev1beta1.Condition, r
 
 	case gardencorev1beta1.ConditionFalse:
 		threshold, ok := b.conditionThresholds[condition.Type]
-		if ok && b.lastOperation != nil && b.lastOperation.State == gardencorev1beta1.LastOperationStateSucceeded && Now().UTC().Sub(b.lastOperation.LastUpdateTime.UTC()) <= threshold {
+		if ok &&
+			((b.lastOperation != nil && b.lastOperation.State == gardencorev1beta1.LastOperationStateSucceeded && Now().UTC().Sub(b.lastOperation.LastUpdateTime.UTC()) <= threshold) ||
+				(reason != condition.Reason || message != condition.Message)) {
```

Comment:
Hmm, what if the message permanently changes?
E.g. the worker controller adds the name of a failed machine to some of the error messages. If for some reason it continuously changes (bug in mcm/worker controller, operation is retried with a new machine or something similar), the error might not be surfaced, although the underlying error doesn't actually change.

Reply:
Can you share such an error message?
I am open to revert it again if we see that it hides too many error that we would actually want to see, no problem.

> I think, we need a bit more sophisticated logic here, unfortunately.

For example?